cmd/cashier/main.go

```diff
@@ -49,22 +50,39 @@ func main() {
 	if err != nil {
 		log.Fatalln("Error generating key pair: ", err)
 	}
-	fmt.Printf("Your browser has been opened to visit %s\n", c.CA)
-	if err := browser.OpenURL(c.CA); err != nil {
+	authURL := c.CA
+	listener := &client.Listener{}
+	if c.AutoToken {
+		listener = client.StartHTTPServer()
+		if listener != nil {
+			authURL = fmt.Sprintf("%s?auto_token=%s",
+				c.CA, url.PathEscape(listener.ReceiverURL))
```

Comment:
Does it make sense to allow a url here? What about just passing a port around instead of a full URL and assuming that the receiver will always be running on localhost? I fear that allowing a full URL could open this to some kind of abuse where a token can be sent somewhere else

Reply:
Yep.